vaadin-select-flow/src/main/java/com/vaadin/flow/component/select/Select.java

```diff
@@ -755,8 +761,19 @@ private void reset() {
         if (isEmptySelectionAllowed()) {
             addEmptySelectionItem();
         }
+
+        final AtomicInteger itemCounter = new AtomicInteger(0);
         getDataProvider().fetch(new Query<>()).map(this::createItem)
-                .forEach(this::add);
+                .forEach(component -> {
+                    add(component);
+                    itemCounter.incrementAndGet();
+                });
+
+        final int newDataSize = itemCounter.get();
+        if (dataSize != newDataSize) {
+            dataSize = newDataSize;
+            fireEvent(new SizeChangeEvent<>(this, newDataSize));
+        }
```

Comment:
This should probably be moved so it fires on beforeClientResponse as will be for DataCommunicator so changed during server roundtrip doesn't fire multiple size change events.

Reply:
Done. I've moved fireSizeEvent to beforeClientResponse invocation. Test mocks and harness copy-pasted from your PR. It will be gentrified within checkbox PRs